### ISSUE TITLE
Package not installed properly in Python 2.7

### ISSUE THREAD

**Issue by benzid-wael** [Bachmann1234/diff_cover#159]
When I run this command `diff-cover coverage.xml` I got `ImportError`, below the traceback for the raised error

```
Traceback (most recent call last):
  File "/home/wael/webapps/bin/diff-cover", line 5, in <module>
    from diff_cover.diff_cover_tool import main
  File "//home/wael/webapps/local/lib/python2.7/site-packages/diff_cover/diff_cover_tool.py", line 18, in <module>
    from diff_cover.report_generator import (
  File "/home/wael/webapps/local/lib/python2.7/site-packages/diff_cover/report_generator.py", line 8, in <module>
    from diff_cover.snippets import Snippet
  File "/home/wael/webapps/local/lib/python2.7/site-packages/diff_cover/snippets.py", line 13, in <module>
    from tokenize import open as openpy
ImportError: cannot import name open
```

which look like that not the proper version of `tokenizer` got installed

However, when I checked your requirements file I didn't find that dependency

Can you update your requirements file?

**Comment by Bachmann1234** [comment]
I think you need to move to an older version of diff cover. As of the latest diff cover dropped python 2.7 support

**Comment by jaraco** [comment]
I think it was a mistake to drop support for Python 2.7 without declaring non-support for that Python version with [PEP 345](https://www.python.org/dev/peps/pep-0345/) Python-Requires. As a result, every downstream packager that pulls in this package using Python 2 will get the unsupported version:

```
~ $ python2 -m pip-run diff-cover -- -c "import diff_cover.snippets"
DEPRECATION: Python 2.7 reached the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 is no longer maintained. pip 21.0 will drop support for Python 2.7 in January 2021. More details about Python 2 support in pip can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
Collecting diff-cover
  Using cached diff_cover-4.0.0-py2.py3-none-any.whl (41 kB)
Collecting jinja2-pluralize
  Using cached jinja2_pluralize-0.3.0-py2.py3-none-any.whl (4.8 kB)
Collecting inflect>=0.2.4
  Using cached inflect-3.0.2-py2.py3-none-any.whl (31 kB)
Collecting pluggy
  Using cached pluggy-0.13.1-py2.py3-none-any.whl (18 kB)
Collecting Jinja2>=2.7.1
  Using cached Jinja2-2.11.2-py2.py3-none-any.whl (125 kB)
Collecting MarkupSafe>=0.23
  Using cached MarkupSafe-1.1.1-cp27-cp27m-macosx_10_6_intel.whl (17 kB)
Collecting importlib-metadata
  Using cached importlib_metadata-1.7.0-py2.py3-none-any.whl (31 kB)
Collecting zipp>=0.5
  Using cached zipp-1.2.0-py2.py3-none-any.whl (4.8 kB)
Collecting pathlib2
  Using cached pathlib2-2.3.5-py2.py3-none-any.whl (18 kB)
Collecting configparser>=3.5
  Using cached configparser-4.0.2-py2.py3-none-any.whl (22 kB)
Collecting contextlib2
  Using cached contextlib2-0.6.0.post1-py2.py3-none-any.whl (9.8 kB)
Processing ./Library/Caches/pip/wheels/91/95/75/19c98a91239878abbc7c59970abd3b4e0438a7dd5b61778335/scandir-1.10.0-cp27-cp27m-macosx_10_14_x86_64.whl
Collecting six
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting pygments
  Using cached Pygments-2.5.2-py2.py3-none-any.whl (896 kB)
Installing collected packages: six, scandir, contextlib2, zipp, pathlib2, configparser, MarkupSafe, importlib-metadata, Jinja2, inflect, pygments, pluggy, jinja2-pluralize, diff-cover
Successfully installed Jinja2-2.11.2 MarkupSafe-1.1.1 configparser-4.0.2 contextlib2-0.6.0.post1 diff-cover-4.0.0 importlib-metadata-1.7.0 inflect-3.0.2 jinja2-pluralize-0.3.0 pathlib2-2.3.5 pluggy-0.13.1 pygments-2.5.2 scandir-1.10.0 six-1.15.0 zipp-1.2.0
WARNING: You are using pip version 20.2; however, version 20.2.3 is available.
You should consider upgrading via the '/System/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python -m pip install --upgrade pip' command.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/var/folders/c6/v7hnmq453xb6p2dbz1gqc6rr0000gn/T/pip-run-jTlLBP/diff_cover/snippets.py", line 13, in <module>
    from tokenize import open as openpy
ImportError: cannot import name open
```

Here's what I recommend. Either:

- Re-release 3.0.1 as 4.0.1 and then release 4.1 or 5.0 declaring the non-support for Python 2, or
- Add the declarations to 4.0.1 and yank 4.0 from PyPI.

Without one of the above options, you're requiring every downstream consumer to maintain a copy and separate declarations for different versions of diff-cover based on Python version.

WDYT?

**Comment by Bachmann1234** [comment]
Yeah, looks like I goofed here. Thanks for catching it

**Comment by Bachmann1234** [comment]
Ok I applied @jaraco 's fixed. Released 4.0.1 and yanked 4.0.0. Sorry for the trouble yall

**Comment by jaraco** [comment]
Works like a charm. Thanks for the super-fast response!
